drivers/video/video_mcux_csi.c

```diff
@@ -198,6 +198,11 @@ static int video_mcux_csi_stream_start(const struct device *dev)
 	const struct video_mcux_csi_config *config = dev->config;
 	struct video_mcux_csi_data *data = dev->data;
 	status_t ret;
+	struct video_buf *vbuf;
+
+	while ((vbuf = k_fifo_get(&data->fifo_out, K_NO_WAIT))) {
```

Comment:
I'm unclear what the goal here is. If a consumer is using the video API correctly, they should queue at least `min_vbuf_count` buffers prior to calling `video_stream_start`, right? So the video pipeline should be able to start. Why wouldn't we simply discard any buffers in the `fifo_out` queue when `video_stream_stop` was called to avoid the issue with dequeuing?

Reply:
Ok, I see why we have this requirement, but it seems like a *software* requirement, not a hardware one. The HAL layer chooses to enqueue all remaining buffers, but we could simply bypass calling `CSI_TransferStop`, and call `CSI_Stop` instead, right? My concern is that a user of the video API won't expect buffers that were previously queued to be reused after the video stream is restarted, will they? We need to follow the video API here.

Reply:
- With the LL CSI driver, bypass CSI_TransferStop will not resolve the problem as low-level driver can still hold an in-progress buffer.

>  My concern is that a user of the video API won't expect buffers that were previously queued to be reused after the video stream is restarted, will they? We need to follow the video API here.

- As my observation, in video API, user is expecting that the buffer is reused. This is because when user call video_stream_stop function, all buffers from the input queue will **be flushed** (https://github.com/zephyrproject-rtos/zephyr/blob/58d503268573d465b6f8f456190930e06eef594f/include/zephyr/drivers/video.h#L574)  to the **output_fifo** (https://github.com/zephyrproject-rtos/zephyr/blob/aefda52e42b599592dcfbcbd359aa8dd6e908994/drivers/video/video_mcux_csi.c#L271) , and user must dequeue these buffers from the output fifo. So, logically, if the user is not dequeuing all the buffers in the output fifo, remaining buffers must be moved back to the input fifo to prevent unexpected behavior.